__tests__/sauce-connect-launcher.test.js

```diff
@@ -0,0 +1,293 @@
+
+    sauceConnectLauncher.download(options, (err) => {
+      expect(err).toBeTruthy();
+      expect(err.message).toEqual(expect.stringContaining("Download failed with status code"));
+
+        // Expected command sequence
+      var expectedSequence = [
+        "Missing Sauce Connect local proxy, downloading dependency",
+        "This will only happen once.",
+        "Invalid response status: 404",
+        "Missing Sauce Connect local proxy, downloading dependency",
+        "This will only happen once."
+      ];
+
+      _.each(log,  (message, i) => {
+        expect(message).toMatch(new RegExp("^" + (expectedSequence[i] || "\\*missing\\*")));
+      });
+
+      done();
+    });
+  });
+
+    
+  it("should work with real credentials", (done) => {
+    sauceConnectLauncher(sauceCreds, (err, sauceConnectProcess) => {
+      expect(err).toBeFalsy();
+      expect(sauceConnectProcess).toBeTruthy();
+      sauceConnectLauncher.kill();
+      expect(sauceCreds.log).toEqual(expect.arrayContaining(["Testing tunnel ready"]));
+      sauceConnectProcess.on("exit", () =>{
+        done();
+      });
+    });
+  });
+
+  it("should execute a provided close callback",  (done) => {
+    sauceConnectLauncher(sauceCreds, (err, sauceConnectProcess) => {
+      expect(err).toBeFalsy();
+      expect(sauceConnectProcess).toBeTruthy();
+      sauceConnectProcess.close(() => {
+        done();
+      });
+    });
+  });
+
+  it("extracts the tunnelId from sc output",  (done) => {
+    sauceConnectLauncher(sauceCreds,  (err, sauceConnectProcess) => {
+      expect(err).toBeFalsy();
+      expect(sauceConnectProcess).toBeTruthy();
+      expect(sauceConnectProcess.tunnelId).toBeTruthy();
+    
+      utils.getTunnels( (err, res, body) => {
+        expect(err).toBeFalsy();
+        expect(res.statusCode).toEqual(200);
+        expect(body).toEqual(expect.arrayContaining([sauceConnectProcess.tunnelId]));
+        sauceConnectProcess.close(done);
+      });
+    });
+  });
+
+  it("closes the open tunnel",  (done) => {
+     
+    sauceConnectLauncher(sauceCreds,  (err, sauceConnectProcess) => {
+      expect(err).toBeFalsy();
+      expect(sauceConnectProcess).toBeTruthy();
+      expect(sauceConnectProcess.tunnelId).toBeTruthy();
+
+      utils.getTunnel(sauceConnectProcess.tunnelId,  (err, res, body) => {
+        expect(err).toBeFalsy();
+        expect(res.statusCode).toBe(200);
+        expect(body.status).toBe("running");
+            
+        sauceConnectProcess.close( () => {
+            // setTimeout(function () { // Wait for tunnel to be terminated
+          utils.getTunnel(sauceConnectProcess.tunnelId,  (err, res, body) => {
+            expect(err).toBeFalsy();
+            expect(res.statusCode).toBe(200);
+            expect(body.status).toBe("terminated");
+            done();
+          });
+            // }, 5000);
+        });
+      });
+    });
+  });
+
+  it.skip("allows to spawn sc detached",  (done) => {
```

Comment:
Why is this skipped? CI should be able to run (and pass) that tests, doesn't it?

Reply:
Hello @johanneswuerbach: CI tests are failing and longer to execute, I'l optimize those tests and push back.

Reply:
HI @johanneswuerbach: That test will no longer be valid as saucelabs is not supporting **v4.3.16**

Here are the logs I get while executing that test.  Let me know your thoughts

<img width="500" alt="Screenshot 2019-07-24 at 21 57 00" src="https://user-images.githubusercontent.com/2499490/61832472-26f68500-ae69-11e9-89b7-aa893fee559d.png">

Reply:
Hi @johanneswuerbach: Converting everything to `mocha` made it a bit tough, So kept the mocha tests along with including jest. We can convert that progressively.